File: lqdrect.pas
program lqdrect;
uses crt;
var     a: array[1..1000,1..300] of 0..1;
        m,n: integer;
        fi,fo: text;

procedure input;
var     i,j: integer;
begin
        readln(fi,m,n);
        for i:= 1 to m do begin
                for j:= 1 to n do read(fi,a[i,j]);
                readln(fi);
        end;
end;

procedure process;
var     res,cnt: int64;
        i,j,k: integer;
begin
        res:= 0;
        for i:= 1 to n-1 do
         for j:= i+1 to n do begin
                cnt:= 0;
                for k:= 1 to m do cnt:= cnt + (a[k,i] and a[k,j]);
                res:= res + cnt*(cnt-1) div 2;
         end;
        //
        writeln(fo,res);
end;

begin
        assign(fi,''); reset(fi);
        assign(fo,''); rewrite(fo);
        //
        input;
        process;
        //
        close(fi);
        close(fo);
end.

File: lqdrect/lqdrect.cpp
#include <stdio.h>

using namespace std;

int m, n;
char a[1001][301];

void input() {
    scanf("%d%d", &m, &n);
    for (int i=1; i<=m; i++) 
        for (int j=1; j<=n; j++) scanf("%d", &a[i][j]);
}

void process() {
    int i, j, k, cnt;
    long long res = 0;
    
    for (i=1; i<n; i++) 
        for (j=i+1; j<=n; j++) {
            cnt = 0;
            //for (k=0; k<m; k++) cnt += a[k][i] & a[k][j];
            for (k=1; k<=m; k++) cnt += a[k][i] & a[k][j];
            res += cnt*(cnt-1)/2;
        }
    printf("%lld\n", res);
}

int main() {
    //freopen("lqdrect.inp", "r", stdin);
    //freopen("lqdrect.out", "w", stdout);
    
    input();
    process();
    return 0;
}
